test/test_testing.py

```diff
@@ -2751,6 +2751,19 @@ def _test_push_and_pull_data(self, original_data, encoding, stream_class):
             received_data = infile.read()
         self.assertEqual(original_data, received_data)
 
+    def test_push_and_pull_larger_file(self):
+        # Intent: to ensure things work appropriately with larger files.
+        # Larger files may be sent/received in multiple chunks; this should help for
+        # checking that such logic is correct.
+        data_size = 2**20
+        original_data = "".join(chr(i % 128) for i in range(data_size)).encode('ascii')
```

Comment:
Would `1024*1024` be a clearer way to specify 1MB than `2**20`? Would `os.urandom(data_size)` work just as well for this use case (and be simpler and more efficient) than the join/range loop?

Reply:
1024*1024: sure
urandom: thanks for the hint here, I like the idea